static/src/javascripts/projects/commercial/modules/sticky-top-banner.js

```diff
@@ -1,250 +1,152 @@
-    };
-
-    var render = function (els, state) {
-        var transitionTimingFunction = 'cubic-bezier(0, 0, 0, .985)';
-        var transitionDuration = '1s';
-
-        els.$header.css({
-            'transition': state.shouldTransition
-                ? ['margin-top', transitionDuration, transitionTimingFunction].join(' ')
-                : '',
-            'margin-top': state.adHeight
+            headerHeight = header.offsetHeight;
+            return topSlot.offsetHeight;
+        })
+        .then(function (currentHeight) {
+            return Promise.all([
+                resizeStickyBanner(currentHeight),
+                onScroll()
+            ]);
         });
+    }
 
-        var pageYOffset = state.scrollCoords[1];
-        var userHasScrolledPastHeader = pageYOffset > state.headerHeight;
+    // The height of the top slot changes on 2 occasions:
+    // 1. when the ad is rendered: the ad slot may be manipulated in many ways
+    //    and its geomatry may change
+    // 2. when the ad is dynamically resized: the creative may send a message
+    //    at any point to signal a change of height. Rubicon ads use a legacy
+    //    version of the message system for handling this
+    //
+    // We also listen for scroll events if we need to, to snap the slot in
+    // place when it reaches the end of the header.
+    function setupListeners() {
+        messenger.register('set-ad-height', onRubiconResize);
+        messenger.register('resize', onResize);
+        if (!config.page.hasSuperStickyBanner) {
+            win.addEventListener('scroll', onScroll);
+        }
+        return trackAdRender(topSlotId).then(onTopAdRendered);
+    }
 
-        els.$adBanner.addClass('sticky-top-banner-ad');
-        els.$adBanner.css({
-            'position': !config.page.hasSuperStickyBanner && userHasScrolledPastHeader ? 'absolute' : 'fixed',
-            'top': !config.page.hasSuperStickyBanner && userHasScrolledPastHeader ? state.headerHeight : '',
-            'height': state.adHeight,
-            // Stop the ad from overflowing while we transition
-            'overflow': state.shouldTransition ? 'hidden' : '',
-            'transition': state.shouldTransition
-                ? ['height', transitionDuration, transitionTimingFunction].join(' ')
-                : ''
+    function setupAnimation() {
+        return fastdom.write(function () {
+            header.classList.add('l-header--animate');
+            stickyBanner.classList.add('sticky-top-banner-ad--animate');
         });
+    }
 
-        var diff = state.adHeight - state.previousAdHeight;
-        var adHeightHasIncreased = diff > 0;
-        if (!state.shouldTransition && adHeightHasIncreased) {
-            // If we shouldn't transition, we want to offset their scroll position
-            var pageXOffset = state.scrollCoords[0];
-            els.window.scrollTo(pageXOffset, pageYOffset + diff);
+    function onTopAdRendered(isRendered) {
+        if (isRendered) {
```

Comment:
do you need to return a resolved `promise` in the else to ensure the `promise.all` resolves..?